testing/trino-product-tests/src/main/java/io/trino/tests/hive/TestHiveCoercion.java

```diff
@@ -362,6 +364,19 @@ private void doTestHiveCoercion(HiveTableDefinition tableDefinition)
                 "varchar_to_smaller_varchar",
                 "id");
 
+        // Document case-sensitivity related behavior for nested fields in hive
+        String hiveValueForCaseChangeField;
+        Predicate<String> isFormat = formatName -> tableName.toLowerCase(Locale.ENGLISH).contains(formatName);
+        if (isFormat.test("rctext") || isFormat.test("textfile")) {
+            hiveValueForCaseChangeField = "\"lower2uppercase\":2";
+        }
+        else if (getHiveVersionMajor() == 3 && isFormat.test("orc")) {
+            hiveValueForCaseChangeField = "\"LOWER2UPPERCASE\":null";
+        }
+        else {
+            hiveValueForCaseChangeField = "\"LOWER2UPPERCASE\":2";
+        }
+
```

Comment:
@findepi 

> in particular, is Hive's behavior dependent on file format in use?

yes, the value changes based on format and version. the behavior with dereference (as in `assertNestedSubFields` ) is tricky too.